src/models/event.js

```diff
@@ -299,6 +310,100 @@ utils.extend(module.exports.MatrixEvent.prototype, {
     },
 
     /**
+     * Check if this event is currently being decrypted.
+     *
+     * @return {boolean} True if this event is currently being decrypted, else false.
+     */
+    isBeingDecrypted: function() {
+        return this._decrypting;
+    },
+
+    /**
+     * Start the process of trying to decrypt this event.
+     *
+     * (This is used within the SDK: it isn't intended for use by applications)
+     *
+     * @internal
+     *
+     * @param {module:crypto} crypto crypto module
+     */
+    attemptDecryption: function(crypto) {
+        if (!crypto) {
+            this._badEncryptedMessage("Encryption not enabled");
+            return;
+        }
+
+        if (!this.isEncrypted()) {
+            throw new Error("Attempt to decrypt event which isn't encrypted");
```

Comment:
> it means something has gone wrong somewhere

I don't agree with this. It only means that we failed to do a conditional check prior to calling this function. This makes the API cumbersome to deal with because there's so many ways you can do it "the wrong way". I would rather that there wasn't a wrong way, eg by nopping if the event is decrypted/doesn't need decryption. I don't think that failing to do this conditional check is a symptom of the entire state machine being corrupt, which I think is what you're implying (in which case failing fast is highly desirable).

I'm happy to punt this for now, but I'd like to re-visit this at some point.